UI/cmake/feature-youtube.cmake

```diff
@@ -1,7 +1,8 @@
 if(YOUTUBE_CLIENTID
    AND YOUTUBE_SECRET
    AND YOUTUBE_CLIENTID_HASH MATCHES "^(0|[a-fA-F0-9]+)$"
-   AND YOUTUBE_SECRET_HASH MATCHES "^(0|[a-fA-F0-9]+)$")
+   AND YOUTUBE_SECRET_HASH MATCHES "^(0|[a-fA-F0-9]+)$"
+   AND TARGET OBS::browser-panels)
```

Comment:
Twitch and YouTube service integrations are directly tied to associated browser docks, so it's a necessary dependency.

Reply:
The YouTube integration doesn't require browser panels for just the account connection, IIRC. Its secondary features, such as Chat, do, but the connection itself does not. Twitch's integration does require browser panels for all parts.

Reply:
That might have been true until the recent update by YouTube, which requires browser docks to be present regardless of whether a service connection exists (the dock is added and enabled by default just by selecting YouTube as a service).

Reply:
IIRC, the YouTube Live Control Room dock does not require being connected via the YouTube integration, so it is technically separate.

Reply:
I think _not_ adding the requirement lead to some breakage if the browser source is not available, but cannot replicate the issue anymore.

Reply:
Normally, adding YouTube API key and secret with Browser disables leads to build issue because of how the new dock was added to CMake (and even OBS).

Reply:
> Normally, adding YouTube API key and secret with obs-browser disabled leads to build issues because of how the new dock was added to CMake (and even OBS).
> 
> Edit: Did not see the last edit.

Yeah I couldn't remember at first, but then went in and tried to compile in a slow VM, took some time to get the error. 😄

Reply:
Ah, I think I see what you mean.